corehq/apps/app_manager/id_strings.py

```diff
@@ -43,11 +43,14 @@ def _regex_union(regexes):
 
 
 REGEXES = []
+REGEX_DEFAULT_VALUES = {}
 
 
-def pattern(*patterns):
-    for pattern in patterns:
-        REGEXES.append(_format_to_regex(pattern))
+def pattern(pattern, default=None):
+    REGEXES.append(_format_to_regex(pattern))
+    if default:
+        REGEX_DEFAULT_VALUES[pattern] = default
+
     return lambda fn: fn
```

Comment:
not saying we need to block on my comments here because this was already behaving this way, but i'm not sure i understand why `pattern` is a decorator, since it doesn't alter the returning function in any way.

also having a decorator alter the state of module-level variables like `REGEXES` and `REGEX_DEFAULT_VALUES` seems a bit risky. if `pattern` were to be used as a decorator on a function in another module, then the values of these module-level variables would be different based on whether that module is imported or not, which could cause different behavior based on how django is being invoked (as a web server, from celery, from a management command, etc.)

Reply:
Can't say I understand why it's a decorator, either. The alternative I'd imagine is making it a function and then calling it from within each currently-decorated method, and naming it with an underscore prefix to discourage importing in other modules. I don't mind doing it, but also don't feel strongly that it's much better. Thoughts?

Reply:
@orangejenny since this already was this way, i think it's fine to leave as is but maybe it would be good to just add a comment on the `pattern` decorator mentioning that it should not be used outside of this module

Reply:
sounds good, will do

Reply:
> The alternative I'd imagine is making it a function and then calling it from within each currently-decorated method, and naming it with an underscore prefix to discourage importing in other modules

Ya. thought of that as an alternative too. Having a decorator like that gives an advantage of it being called just once on environment load but having it within the method would have it called each time the parent method is called.